hazelcast-wm/src/main/java/com/hazelcast/web/ClusteredSessionService.java

```diff
@@ -0,0 +1,567 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.web;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import com.hazelcast.map.EntryBackupProcessor;
+import com.hazelcast.map.EntryProcessor;
+import com.hazelcast.map.impl.MapEntrySimple;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.nio.serialization.SerializationService;
+import com.hazelcast.spi.impl.SerializationServiceSupport;
+import com.hazelcast.util.EmptyStatement;
+
+import javax.servlet.FilterConfig;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Queue;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+
+public class ClusteredSessionService {
+
+    protected static final ILogger LOGGER = Logger.getLogger(ClusteredSessionService.class);
+    private static final long CLUSTER_CHECK_INTERVAL = 1000L;
+    private static final long RETRY_MILLIS = 7000;
+    private final String jvmId = UUID.randomUUID().toString();
+    private volatile IMap clusterMap;
+    private volatile SerializationServiceSupport sss;
+    private volatile HazelcastInstance hazelcastInstance;
+
+    private final FilterConfig filterConfig;
+    private final Properties properties;
+    private final String clusterMapName;
+    private final String sessionTTL;
+
+    private final Queue<String> orphanSessions = new LinkedBlockingQueue<String>();
+    private volatile boolean failedConnection = true;
+    private volatile long lastConnectionTry;
+    private final ExecutorService es = Executors.newSingleThreadExecutor();
+
+
+
+    public ClusteredSessionService(FilterConfig filterConfig, Properties properties, String clusterMapName, String sessionTTL) {
+        this.filterConfig = filterConfig;
+        this.properties = properties;
+        this.clusterMapName = clusterMapName;
+        this.sessionTTL = sessionTTL;
+        es.execute(new Runnable() {
+            @Override
+            public void run() {
+                while (true) {
```

Comment:
@mesutcelik If ```shutdown-on-destroy``` parameter of hazelcast filter is true the loop will break ```es``` will shutdown with hazelcast instance. if it is false,  ```es``` will execute ```ensureInstance``` method every one second, because the threads created by the executor framework are not managed by application server. Is this the desired behaviour, or we need to shutdown ```es```  regardless of the value of ```shutdown-on-destroy```?